gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,385 @@
+            if error:
+                return error
+
+        guppy = exp_config.get("guppyConfig")
+        if not guppy:
+            return FieldSyntaxError("explorerConfig.guppyConfig")
+        if not guppy.get("dataType"):
+            return FieldSyntaxError("explorerConfig.guppyConfig.dataType")
+
+        manifest_mapping = guppy.get("manifestMapping")
+        buttons = exp_config.get("buttons")
+        val_mapping = False
+        for button in buttons:
+            if button.get("enabled") and button.get("type") == "manifest":
+                val_mapping = True
+        if manifest_mapping and val_mapping:
+            checks = [
+                "resourceIndexType",
+                "resourceIdField",
+                "referenceIdFieldInResourceIndex",
+                "referenceIdFieldInDataIndex",
+            ]
+            error = check_required_fields(
+                "explorerConfig.guppyConfig.manifestMapping", checks, manifest_mapping
+            )
+            if error:
+                return error
+
+    study_viewer = gitops.get("studyViewerConfig")
+    if study_viewer:
+        checks = ["dataType", "listItemConfig", "rowAccessor"]
+        for viewer in study_viewer:
+            error = check_required_fields("studyViewerConfig", checks, viewer)
+            if error:
+                return error
+    return None
+
+
+def check_required_fields(path, checks, field):
+    for check in checks:
+        if not field.get(check):
+            return FieldSyntaxError(f"{path}.{check}")
+    return None
+
+
+def check_field_value(path, checks, accepted_values, errors):
+    for check in checks:
+        if check not in accepted_values:
+            errors.append(FieldError("Invalid field {} in {}".format(check, path)))
+    return errors
+
+
+def _validate_itemConfig(item_config, props, errors):
+    errors = check_field_value(
+        "studyViewerConfig.blockFields.(list/single)ItemConfig",
+        item_config.get("blockFields", []),
+        props,
+        errors,
+    )
+    errors = check_field_value(
+        "studyViewerConfig.tableFields.(list/single)ItemConfig",
+        item_config.get("tableFields", []),
+        props,
+        errors,
+    )
+    return errors
+
+
+def _validate_studyViewerConfig_helper(viewer, type_prop_map, errors):
+    datatype = viewer.get("dataType")
+    props = type_prop_map.get(datatype, [])
+    if not props:
+        errors.append(
+            FieldError(
+                "Invalid field {} in studyViewerConfig.dataType".format(datatype)
+            )
+        )
+
+    listitemconfig = viewer.get("listItemConfig")
+    errors = _validate_itemConfig(listitemconfig, props, errors)
+    if viewer.get("singleItemConfig"):
+        errors = _validate_itemConfig(viewer.get("singleItemConfig"), props, errors)
+
+    for dtype, props in type_prop_map.items():
+        if viewer["rowAccessor"] not in props:
```

Comment:
`rowAccessor` doesn't have to be in _all_ indices, it has to be in index `dataType`, and in indices `fileDataType` and `docDataType` if they're used